rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -8,23 +8,212 @@
 
 LOG_CHANNEL(cellMic);
 
+template <>
+void fmt_class_string<microphone_handler>::format(std::string& out, u64 arg)
+{
+	format_enum(out, arg, [](auto value)
+	{
+		switch (value)
+		{
+		case microphone_handler::null: return "Null";
+		case microphone_handler::standard: return "Standard";
+		case microphone_handler::singstar: return "Singstar";
+		}
+
+		return unknown;
+	});
+}
+
+void mic_context::variable_byteswap(u8* src, u8* dst, u32 bytesize)
+{
+	switch (bytesize)
+	{
+	case 4:
+		dst[0] = src[3];
```

Comment:
Dont byteswap by hand, src and dst can be void* for easier casting.
```suggestion
		*(u32*)dst = *(be_t<u32>*)src;
```